clock_exercise/clockmanager.h

```diff
@@ -7,40 +7,72 @@ class ClockManager : public QObject
 {
     Q_OBJECT
 
+    // ↓↓↓ properties
     Q_PROPERTY(QString dateText MEMBER m_dateText NOTIFY dateTextChanged)
     Q_PROPERTY(int minutes MEMBER m_minutes NOTIFY minutesChanged)
     Q_PROPERTY(int hours MEMBER m_hours NOTIFY hoursChanged)
-    Q_PROPERTY(int timeOfTheDissolveAnimation MEMBER TIME_OF_THE_DISSOLVE_ANIMATION CONSTANT)
     Q_PROPERTY(bool timerRunning MEMBER m_timerRunning NOTIFY timerRunningChanged)
     Q_PROPERTY(float timerCurrentSeconds MEMBER m_timerCurrentSeconds NOTIFY timerCurrentSecondsChanged WRITE setTimerCurrentSeconds)
+    Q_PROPERTY(int dayOfDate MEMBER m_dayOfDate NOTIFY dayOfDateChanged WRITE setDayOfDate)
+    Q_PROPERTY(int monthOfDate MEMBER m_monthOfDate NOTIFY monthOfDateChanged WRITE setMonthOfDate)
+    Q_PROPERTY(int yearOfDate MEMBER m_yearOfDate NOTIFY yearOfDateChanged WRITE setYearOfDate)
+    Q_PROPERTY(QString dayOfDateName MEMBER m_dayOfDateName NOTIFY dayOfDateNameChanged WRITE setDayOfDateName)
+    Q_PROPERTY(QString monthOfDateName MEMBER m_monthOfDateName NOTIFY monthOfDateNameChanged WRITE setMonthOfDateName)
     Q_PROPERTY(int secondsInOneMinute MEMBER SECONDS_IN_ONE_MINUTE CONSTANT)
     Q_PROPERTY(int minutesInOneHour MEMBER MINUTES_IN_ONE_HOUR CONSTANT)
     Q_PROPERTY(int hoursInADay MEMBER HOURS_IN_A_DAY CONSTANT)
+    Q_PROPERTY(int timeOfTheDissolveAnimation MEMBER TIME_OF_THE_DISSOLVE_ANIMATION CONSTANT)
+    Q_PROPERTY(int alarmDay MEMBER m_alarmDay NOTIFY alarmDayChanged WRITE setAlarmDay)
+    Q_PROPERTY(int alarmMonth MEMBER m_alarmMonth NOTIFY alarmMonthChanged WRITE setAlarmMonth)
+    Q_PROPERTY(int alarmYear MEMBER m_alarmYear NOTIFY alarmYearChanged WRITE setAlarmYear)
```

Comment:
Non capisco in cosa differiscano queste property da quelle definite sopra (`dayOfDate` etc.). Mi sembra che indichino la stessa quantità o sbaglio?

Reply:
Le une rappresentavano la data della pagina di settaggio allarme e le altre rappresentavano la data della pagina di settaggio data, dovevano essere diverse perché quando dalla pagina di settaggio allarme si passava alla pagina di settaggio data, la data della pagina di settaggio allarme doveva rimanere memorizzata nel caso dovesse essere ripristinata a causa di un back, adesso la data della pagina di settaggio data è stata portata su qml, quindi nella parte c++ rimane solo la data della pagina di settaggio allarme.